mirascope/core/openai/call_response_chunk.py

```diff
@@ -87,3 +92,23 @@ def output_tokens(self) -> int | None:
         if self.usage:
             return self.usage.completion_tokens
         return None
+
+    @computed_field
+    @property
+    def _audio(self) -> dict[str, Any] | None:
+        """Returns the audio of the response."""
+        return getattr(self.chunk.choices[0].delta, "audio", None)
+
+    @property
+    def audio(self) -> bytes | None:
+        """Returns the audio data of the response."""
+        if (audio := self._audio) and (audio_data := audio.get("data")):
+            return base64.b64decode(audio_data)
+        return None
+
+    @property
+    def audio_transcript(self) -> str | None:
+        """Returns the transcript of the audio content."""
+        if audio := self._audio:
+            return audio.get("transcript")
+        return None
```

Comment:
OpenAI SDK didn't define `audio` field in `ChoiceDelta`.
https://github.com/openai/openai-python/blob/fc4fce30aff7620c32e57301d3c40b4883810825/src/openai/types/chat/chat_completion_chunk.py#L59-L78
But, actual response has `audio` dict.
So, I added audio and audio_transcript in `OpenAICallResponseChunk` by using dict().get()

The field is not even defined in their openapi.
If you should not use non-public objects, remove this one.

https://github.com/openai/openai-python/blob/main/.stats.yml

Reply:
hmm, have you tested that it actually streams the audio back? if it does, then I think this is fine?

same thing here though, update the `audio` and `audio_transcript` properties as mentioned in my comment on the standard call response

Reply:
Yes, I tested this code. It works fine.

### Code:
```python
import io
import wave

from pydub.playback import play

from mirascope.core import Messages, openai
from mirascope.core.openai import AudioSegment


@openai.call(
    "gpt-4o-audio-preview",
    call_params={"audio": {"voice": "alloy", "format": "pcm16"}, "modalities": ["text", "audio"],},
    stream=True)
def recommend_book(genre: str) -> Messages.Type:
    return f"Recommend a {genre} book"

audio_chunk = b''
audio_transcript_chunk = ''
for chunk, _ in recommend_book("fantasy"):
    if chunk.audio:
        audio_chunk += chunk.audio
    if chunk.audio_transcript:
        audio_transcript_chunk += chunk.audio_transcript

print(audio_transcript_chunk)

SAMPLE_WIDTH = 2
FRAME_RATE = 24000
CHANNELS = 1

audio_segment = AudioSegment.from_raw(
        io.BytesIO(audio_chunk),
        sample_width=SAMPLE_WIDTH,
        frame_rate=FRAME_RATE,
        channels=CHANNELS,
    )
play(audio_segment)
```

### Outputs:
```bash
I highly recommend "The Name of the Wind" by Patrick Rothfuss. It's a beautifully written novel about a young man named Kvothe who grows up to become a legendary figure in the world of magic. With rich world-building and memorable characters, it's the first book in a planned trilogy called "The Kingkiller Chronicle."
Input #0, wav, from '/var/folders/1c/6xfc8h09781fp0zdndc007xh0000gn/T/tmpuk8lql66.wav':
  Duration: 00:00:22.55, bitrate: 384 kb/s
  Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 24000 Hz, 1 channels, s16, 384 kb/s
```